test/functional/mempool_limit.py

```diff
@@ -20,55 +23,55 @@ def set_test_params(self):
-        relayfee = self.nodes[0].getnetworkinfo()['relayfee']
+        node = self.nodes[0]
+        miniwallet = MiniWallet(node)
+        relayfee = node.getnetworkinfo()['relayfee']
 
-        self.log.info('Check that mempoolminfee is minrelytxfee')
-        assert_equal(self.nodes[0].getmempoolinfo()['minrelaytxfee'], Decimal('0.00001000'))
-        assert_equal(self.nodes[0].getmempoolinfo()['mempoolminfee'], Decimal('0.00001000'))
+        self.log.info('Check that mempoolminfee is minrelaytxfee')
+        assert_equal(node.getmempoolinfo()['minrelaytxfee'], Decimal('0.00001000'))
+        assert_equal(node.getmempoolinfo()['mempoolminfee'], Decimal('0.00001000'))
 
-        txids = []
-        utxos = create_confirmed_utxos(self, relayfee, self.nodes[0], 91)
+        tx_batch_size = 25
+        num_of_batches = 3
+        # Generate UTXOs to flood the mempool
+        # 1 to create a tx initially that will be evicted from the mempool later
+        # 3 batches of multiple transactions with a fee rate much higher than the previous UTXO
+        # And 1 more to verify that this tx does not get added to the mempool with a fee rate less than the mempoolminfee
+        self.generate(miniwallet,1 + (num_of_batches * tx_batch_size) + 1)
+
+        # Mine 99 blocks so that the UTXOs are allowed to be spent
+        self.generate(node,COINBASE_MATURITY - 1)
 
         self.log.info('Create a mempool tx that will be evicted')
-        us0 = utxos.pop()
-        inputs = [{ "txid" : us0["txid"], "vout" : us0["vout"]}]
-        outputs = {self.nodes[0].getnewaddress() : 0.0001}
-        tx = self.nodes[0].createrawtransaction(inputs, outputs)
-        self.nodes[0].settxfee(relayfee) # specifically fund this tx with low fee
-        txF = self.nodes[0].fundrawtransaction(tx)
-        self.nodes[0].settxfee(0) # return to automatic fee selection
-        txFS = self.nodes[0].signrawtransactionwithwallet(txF['hex'])
-        txid = self.nodes[0].sendrawtransaction(txFS['hex'])
-
-        relayfee = self.nodes[0].getnetworkinfo()['relayfee']
-        base_fee = relayfee*100
-        for i in range (3):
-            txids.append([])
-            txids[i] = create_lots_of_big_transactions(self.nodes[0], txouts, utxos[30*i:30*i+30], 30, (i+1)*base_fee)
+        tx_to_be_evicted_id = miniwallet.send_self_transfer(from_node=node, fee_rate=relayfee)["txid"]
+
+        # Increase the tx fee rate massively to give the subsequent transactions a higher priority in the mempool
+        base_fee = relayfee * 1000
+
+        self.log.info('Fill up the mempool with txs with higher fee rate')
+        num_of_large_tx_sent = 0
+        txouts = gen_return_txouts()
+
+        for batch_of_txid in range(num_of_batches):
+            # Increment the tx fee rate gradually by a factor of (base_fee) for each batch
+            miniwallet.send_large_transactions(node, txouts, tx_batch_size, (batch_of_txid+1) * base_fee)
+            num_of_large_tx_sent +=tx_batch_size
```

Comment:
this should be 
```suggestion
            num_of_large_tx_sent += tx_batch_size
```
per https://www.python.org/dev/peps/pep-0008/#other-recommendations

Reply:
All the styling changes have been handled in e8a3ff3b586beb4e2bb2d77a3bf4d7b98a1a2df2 :))

Hope this answers the query

Reply:
ah, gotcha, i missed that in the second commit.

in general, I would avoid touching the same line multiple times in different commits as it requires the reviewer to look at the same line multiple times. not a bid deal tho

Reply:
It is a bit confusing to add new code in the wrong style and then fix it up in the second commit. Why not add it with the right style from the beginning?